tools/merge-api-changelogs

```diff
@@ -0,0 +1,115 @@
+#!/usr/bin/env python3
+import fileinput
+import glob
+import os
+import re
+import subprocess
+import sys
+
+
+def get_changelog_files_list() -> list[str]:
+    dir_path = "api_docs/unmerged.d"
+    if os.path.exists(dir_path):
+        return [os.path.basename(path) for path in glob.glob(f"{dir_path}/ZF-??????")]
+
+    return []
+
+
+def get_unmerged_changelogs() -> str:
+    changelogs = ""
+    dir_path = "api_docs/unmerged.d"
+    changelog_files_list = get_changelog_files_list()
+    if changelog_files_list:
+        print(f"Unmerged changelog files: {changelog_files_list}")
+    else:
+        print("No unmerged changelog files found.")
+
+    for file_name in changelog_files_list:
+        file_path = f"{dir_path}/{file_name}"
+        with open(file_path) as f:
+            changelogs += f.read()
+
+    return changelogs
+
+
+def increment_and_get_feature_level() -> int:
+    new_feature_level = None
+    with fileinput.FileInput("version.py", inplace=True) as file:
+        for line in file:
+            if line.startswith("API_FEATURE_LEVEL = "):
+                match = re.search(r"\d+", line)
+                if match:
+                    new_feature_level = int(match.group()) + 1
+                    print(f"API_FEATURE_LEVEL = {new_feature_level}")
+                    continue
+
+            print(line, end="")
+
+    assert new_feature_level is not None
+    print(f"Updated API feature level: {new_feature_level - 1} -> {new_feature_level}")
+    return new_feature_level
+
+
+def merge_changelogs(changelogs: str, new_feature_level: int) -> None:
+    changlogs_merged = False
+    changelog_path = "api_docs/changelog.md"
+    with fileinput.FileInput(changelog_path, inplace=True) as file:
+        for line in file:
+            print(line, end="")
+            if not changlogs_merged and bool(re.fullmatch(r"## Changes in Zulip \d+\.\d+\n", line)):
+                changlogs_merged = True
+                print(f"\n**Feature level {new_feature_level}**\n")
+                print(changelogs, end="")
+
+    print(f"Changelogs merged to {changelog_path}.")
+
+
+def update_feature_level_in_api_docs(new_feature_level: int) -> None:
+    changelog_files_list = get_changelog_files_list()
+    num_replaces = 0
+    api_docs_path = "zerver/openapi/zulip.yaml"
+    with fileinput.FileInput(api_docs_path, inplace=True) as file:
+        for line in file:
+            for file_name in changelog_files_list:
```

Comment:
Let's write explicitly `temporary_feature_level = file_name` to make it explicit.